### PR TITLE
Passing the PYTHONPATH to the child process.

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

When using sys.path.append or sys.path.insert doesn't seem to work with the simulator. I was able to reproduce the same issue also using just a python script, i.e. running python test.py. I cannot import the required modules inside the simulator run. However using export PYTHONPATH before running the script works fine. Interestingly, importing the module outside the SimulatorRunner works fine. The sys.path is not correctly passed on to subprocesses created by the simulator.

### Description

Pass the PYTHONPATH to the child process for SImulatorRunner.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/app/simulator/simulator_runner.py [hunk 1]
```diff
@@ -532,7 +532,9 @@ def do_one_task(self, client, num_of_threads, gpu, lock, timeout=60.0):
         )
         if gpu:
             command += " --gpu " + str(gpu)
-        _ = subprocess.Popen(shlex.split(command, True), preexec_fn=os.setsid, env=os.environ.copy())
+        new_env = os.environ.copy()
+        new_env["PYTHONPATH"] = ":".join(sys.path)[1:]
```
Comment (holgerroth) [inline nvflare/private/fed/app/simulator/simulator_runner.py:536]: Is this linux specific? Could it be done so different platforms are supported?

Comment (yhwen) [reply nvflare/private/fed/app/simulator/simulator_runner.py]: Changed to use os.pathsep to work platform independent.

Comment (yanchengnv) [reply nvflare/private/fed/app/simulator/simulator_runner.py]: Don't understand what problem this PR tries to solve.
Why should we explicitly add everything on sys.path to PYTHONPATH?

File: nvflare/private/fed/app/simulator/simulator_runner.py [hunk 2]
```diff
@@ -532,7 +532,10 @@ def do_one_task(self, client, num_of_threads, gpu, lock, timeout=60.0):
         )
         if gpu:
             command += " --gpu " + str(gpu)
-        _ = subprocess.Popen(shlex.split(command, True), preexec_fn=os.setsid, env=os.environ.copy())
+        new_env = os.environ.copy()
+        new_env["PYTHONPATH"] = os.pathsep.join(sys.path)[1:]
+        os.
```
Comment (yanchengnv) [inline nvflare/private/fed/app/simulator/simulator_runner.py:537]: This is syntax error.

**Review by yanchengnv — CHANGES_REQUESTED** [review]
There is syntax error.

File: nvflare/private/fed/app/simulator/simulator_runner.py (hunk 2 above)
Comment (yhwen) [reply nvflare/private/fed/app/simulator/simulator_runner.py]: fixed the typo.

File: nvflare/private/fed/app/simulator/simulator_runner.py (hunk 1 above)
Comment (yhwen) [reply nvflare/private/fed/app/simulator/simulator_runner.py]: Holger and one customer reported an issue running Simulator. 

When using sys.path.append or sys.path.insert doesn't seem to work with the simulator. I was able to reproduce the same issue also using just a python script, i.e. running python test.py. I cannot import the required modules inside the simulator run. However using export PYTHONPATH before running the script works fine. Interestingly, importing the module outside the SimulatorRunner works fine. The sys.path is not correctly passed on to subprocesses created by the simulator.

File: nvflare/private/fed/app/simulator/simulator_runner.py [hunk 3]
```diff
@@ -532,7 +532,9 @@ def do_one_task(self, client, num_of_threads, gpu, lock, timeout=60.0):
         )
         if gpu:
             command += " --gpu " + str(gpu)
-        _ = subprocess.Popen(shlex.split(command, True), preexec_fn=os.setsid, env=os.environ.copy())
+        new_env = os.environ.copy()
+        new_env["PYTHONPATH"] = os.pathsep.join(sys.path)[1:]
```
Comment (holgerroth) [inline nvflare/private/fed/app/simulator/simulator_runner.py:536]: Why do we need to index with `[1:]`?

**Review by holgerroth — APPROVED** [review]
Looks good. It works when tested in my environment.

**Review by holgerroth — APPROVED** [review]

File: nvflare/private/fed/app/simulator/simulator_runner.py [hunk 4]
```diff
@@ -532,7 +532,12 @@ def do_one_task(self, client, num_of_threads, gpu, lock, timeout=60.0):
         )
         if gpu:
             command += " --gpu " + str(gpu)
-        _ = subprocess.Popen(shlex.split(command, True), preexec_fn=os.setsid, env=os.environ.copy())
+        new_env = os.environ.copy()
+        if not sys.path[0]:
+            new_env["PYTHONPATH"] = os.pathsep.join(sys.path[1:])
```
Comment (holgerroth) [inline nvflare/private/fed/app/simulator/simulator_runner.py:537]: Shouldn't is be `os.pathsep.join(sys.path)[1:]`

**Review by holgerroth — APPROVED** [review]

**Pull request merged**